src/mm2_common.h

```diff
@@ -25,12 +25,15 @@ Multiple declarations will cause compiler errors!
 // Allocator
 // Calls MM2's internal operator new
 //
-#define ANGEL_ALLOCATOR void* operator new(size_t size)          { return hook::StaticThunk<0x577360>::Call<void*>(size); } \
-                        void operator delete(void * pointer)     { hook::StaticThunk<0x577380>::Call<void>(pointer);}       \
-                        void operator delete[](void * pointer)   { hook::StaticThunk<0x5773C0>::Call<void>(pointer);}
+#define ANGEL_ALLOCATOR void* operator new(size_t size)                        { return hook::StaticThunk<0x577360>::Call<void*>(size); } \
+                        void operator delete(void * pointer)                   { hook::StaticThunk<0x577380>::Call<void>(pointer);} \
+                        void operator delete[](void * pointer)                 { hook::StaticThunk<0x5773C0>::Call<void>(pointer);} \
+
+#define LEVEL_ALLOCATOR void* operator new(size_t size)                        { return hook::StaticThunk<0x463110>::Call<void*>(size); } \
+                        void operator delete(void * pointer)                   { hook::StaticThunk<0x463170>::Call<void>(pointer);} \
+                        void* operator new(size_t size, void * data)           { return data; } \
+                        void operator delete(void * pointer, void * place)     { place; }
```

Comment:
I'm not entirely sure what the point of this `void operator delete(void * pointer, void * place) { place; }` line is, but if this is correct, at least do `(void)place;`.

Reply:
I had to match delete operator with new operator otherwise memory won't be freed if initialization throws an exception.